aristanetworks/ansible-avd
# Feat(eos_cli_config_gen): Add event-handler trigger on-maintenance.

## Change Summary

Add event-handler trigger "on-maintenance".

## Related Issue(s)

Fixes #2951 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

Adding an option for "on-maintenance" in event-handlers.trigger and updated the data-model for the same.

## How to test

Add input trigger on-maintenance:
```
- name: trigger-on-maintenance
    trigger: on-maintenance
    trigger_on_maintenance:
      operation: enter
      interface:
        interface_type: Ethernet
        interface_number: 4
      stage: all
    action_type: bash
    action: echo "on-maintenance"
```

Check the output config:

```
!
event-handler trigger-on-maintenance
   trigger on-maintenance enter interface Ethernet 4 all
   action bash echo "on-maintenance"
```
Test the config on EOS CLI

## Checklist

### Repository Checklist
